src/lib/markbind/src/patches/htmlparser2.js

```diff
@@ -1,4 +1,23 @@
-const { Tokenizer } = require('htmlparser2');
+/*
+ * Four behaviours of htmlparser2 are patched here, the first 2 being 'convenience' patches
+ * to avoid repeated passing of lowerCaseAttributeNames and recognizeSelfClosing options.
+ * 1. Defaulting to self closing tag recognition
+ * 2. Disabling automatic attribute name conversion to lower case
+ * 3. Recognising text in markdown inline code / code blocks as raw text
+ * 4. Ability to inject/whitelist certain tags to be parsed like script/style tags do. (special tags)
+ */
+
+const { Tokenizer, Parser } = require('htmlparser2');
+
+// Self closing tag recognition patch
+Parser.prototype.onselfclosingtag = function () {
```

Comment:
https://github.com/fb55/htmlparser2/wiki/Parser-options

Enables `recognizeSelfClosing` and disables `lowerCaseAttributeNames` so there isn't a need to keep passing in those options.
Not sure if passing it in may be better on the other hand though, in the case of a new developer very familiar with "anonymous self closing tags" in `html`, they would have to find this file and the comment above if they're wondering why it behaves as such

Reply:
I think we can add some comments here to provide the context as to why we are patching this. I think it makes sense to simplify usage at the call site, and this file is the natural first place to look if anyone is wondering why the behaviour might not be as they expected.